index.html

```diff
@@ -696,14 +697,101 @@
+        var pfkeys;
+        /* First, let's check that ECDH is supported. We'll need that
+         * for pretty much everything.
+         */
+        webCrypto.generateKey(
+             {name:"ECDH", namedCurve: "P-256"},
+             true,
+             ["deriveBits"])
+            .then(new_keys => {
+                /* Neat, now on to 'raw' key format. Raw keys are used
+                 * for things like the Crypto Key values. It's possible to 
+                 * fake them (see common.js JWKToRaw) but there are lots
+                 * of potential bugs in that approach. 
+                 */
+                pfkeys = new_keys;
+                webCrypto.exportKey('raw', pfkeys.publicKey)
+                .then(key => {
+                    /* Can we import a public key for key bit derivation?
+                     * This is used by signing mechanisms like VAPID and 
+                     * data sends. Again, possible to fake, but introduces
+                     * more bugs than it solves.
+                     */
+                    webCrypto.importKey(
+                        'raw',
+                        key,
+                        {name:"ECDH", namedCurve:"P-256"},
+                        true,
+                        ['deriveBits'])
+                        .then( key => {
+                            /* And finally, can we derive bits from the 
+                             * imported key and the private key? This is 
+                             * the core of ECDSA. If it fails, yeah, nothing
+                             * is ever going to work. Christmas is ruined 
+                             * forever, etc.
+                             */
+                            webCrypto.deriveBits(
+                                {name:"ECDH",
+                                 namedCurve:"P-256",
+                                 public: key},
+                                pfkeys.privateKey,
+                                256)
+                            .then(derived => {
+                                console.debug("Preflight is Ok");
+                                return true;
+                                })
+                            .catch(err => {
+                                show_err(msg + 
+                                "<br>ECDSA key derivation not supported.<br>" +
+                                err.toString());
+                                console.error(err);
+                                });
+                            })
+                        .catch(err => {
+                            show_err(msg + 
+                                "<br>Raw key import not supported.</br>" +
+                                  err.toString());
+                            console.error(err); 
+                        });
+                })
+                .catch(err => {
+                    show_err(msg + 
+                        "<br>Raw key export not supported.<br>" +
```

Comment:
Include link to MDN? https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/exportKey#Browser_compatibility